### PR TITLE
[WIP] Zwave panel

### PR DESCRIPTION
**Pull request by turbokongen**
For https://github.com/home-assistant/home-assistant/pull/7127
This is an open PR so anybody can check in and comment. By no means finished or polished.
Services available:
 - [x] Add Node(secure)
 - [x] Remove Node
 - [x] Cancel command
 - [x] Heal Network
 - [x] Start network
 - [x] Stop Network
 - [x] Soft reset
 - [x] Test network
 - [x] Set config parameter
 - [x] Print config parameter
 - [x] Remove failed node
 - [x] Replace failed node
 - [x] Change association
 - [x] Set Wakeup
 - [x] Print node
 - [x] Refresh entity
 - [x] Refresh node
 - [x] view of OZW_Log.txt
 - [x] node group members
 - [x] Node information (zwave device entity attributes)
 - [x] set_usercode
 - [x] clear_usercode
image

### CODE REVIEW

**Comment by balloob** [comment]
Turbo, a 1000 lines component ?! Can you please break it up. Preferably just do 1 card per component.

**Comment by balloob** [comment]
Also MVP haha, next time we should do 1 feature at a time. Makes code review a lot easier

File: panels/zwave/ha-panel-zwave.html [hunk 1]
```diff
@@ -91,7 +112,319 @@
...
+            </paper-dropdown-menu>
+          </div>
+
+          <template is='dom-if' if='[[!computeIsNodeSelected(selectedNode)]]'>
+          <div class='card-actions'>
+            <ha-call-service-button
+                hass='[[hass]]'
+                domain='zwave'
+                service='refresh_node'
+                service-data=[[computeNodeServiceData(selectedNode)]]
```
Comment (balloob) [inline panels/zwave/ha-panel-zwave.html]: If you're going to compute the same service data many times, put it in a computed property so it is only computed once.

**Comment by turbokongen** [comment]
So, I have exported all cards, leaving only  the main node card, reducing the main panel to 433 lines.

**Comment by balloob** [comment]
Thanks @turbokongen . I will scan over it but won't go in depth on the review, ain't nobody got time for that

**Review by balloob — APPROVED** [review]
It looks all good. 🐬  Super stoked for this! 

Ok to merge when logging statements have been removed and linting addressed https://travis-ci.org/home-assistant/home-assistant-polymer/builds/231836811#L247

File: panels/zwave/ha-panel-zwave.html [hunk 2]
```diff
@@ -112,6 +225,209 @@
...
+    if (this.selectedNode === -1 ||
+      !this.nodes[selectedNode].entity_id) return -1;
+    var str = (this.nodes[selectedNode].entity_id);
+    var name = str.replace('zwave.', '');
+//    console.log('Node name: ', name);
+    return name;
+  },
+
+  computeNodeNameServiceData: function (newNodeNameInput) {
+//    console.log('Rename node:', JSON.stringify({ node_id: this.nodes[this.selectedNode].attributes.node_id,
```
Comment (balloob) [inline panels/zwave/ha-panel-zwave.html]: For the future, you don't want to stringify before logging. That way you can actually play with the object.

**Comment by turbokongen** [comment]
Wow! Really? I was super prepared for a good piece of work with review changes. :smiley: 
Anyway, I have a little quirk with the set-config service, so I  am reworking how it is being used in the backend. The value is different than the content, so It needs to be set as string on list parameters. As son as I have it working and tests are modifyed I'll submit a PR for the backend for that, and clean up the frontend code. Then we are ready to go!!! It's been a good piece of work.

**Pull request merged**